src/native/managed/cdacreader/src/Data/MethodTableArray.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Diagnostics.DataContractReader.Contracts;
+
+namespace Microsoft.Diagnostics.DataContractReader.Data;
+
+internal sealed class MethodTableArray : IData<MethodTableArray, (TargetPointer ptr, int size)>
```

Comment:
this doesn't feel like it belongs in `Data`.  I've typically viewed `Data` as straightforward data-descriptor extractor.  Something we could potentially source-generate in the future.  This `MehtodTableArray` feels more like it belongs to the RuntimeTypeSystem contract proper

Reply:
I think we need a general structure for post-processed data produced by the contracts. @lambdageek you have already done this with the _methodTables array in the `RuntimeTypeSystem` contract, and I think that is a better place to put the processing, but I don't think its a particulaly good way to structure managing the memory of the data.

As I was writing up the `DacStreams` contract for the pointer to name mapping, I think I found a better approach.

1. Use the `ProcessedData`/`IData` interfaces to hold the data. I dislike the `RuntimeTypeSystem` approach of having some state in the actual contract implementation. I want ALL state for the contracts to be held in `ProcessedData`, or on other structures associated with the Target (such as the `Metadata`).
2. Put the data structure into the contract implementation.

This both lets us have a single part of the system to do all memory invalidation, lets us get the caching behavior to avoid reprocessing the world all of the time, and places all of the logic that understands runtime data structures into the contract.